framework/include/constraints/MortarConstraintBase.h

```diff
@@ -150,6 +150,15 @@ class MortarConstraintBase : public Constraint,
   /// The shape function gradients corresponding to the primary interior primal variable
   const VariableTestGradient & _grad_test_primary;
 
+  /// the higher-dimensional secondary face element
+  const Elem * const & _interior_secondary_elem;
+
+  /// the higher-dimensional primary face element
+  const Elem * const & _interior_primary_elem;
+
+  /// the lower-dimensional secondary element
+  const Elem * const & _lower_secondary_elem;
+
```

Comment:
why does the whole file go secondary before primary

Reply:
that would have been my comment

Reply:
mortar terminology is seemingly very backwards. Traditionally what is now the "secondary" is known in the literature as "slave" ... which is stupid because the "slave/secondary" basically runs the show. There are often times when primary data is not needed at all ... and hence has been added after secondary data in many of these classes which has lead to the data ordering structure in the classes